### DISCUSSION TITLE
Attachments downloads instead of opening in a new tab

### DISCUSSION THREAD

**Discussion by dkrut** [allure-framework/allure2#2141] [General Discussion]
### Describe the Bug

After update allure from 2.23.1 to 2.24.0 changed behaviour opening attachments in report. When click to open attachment in new tab, download window opens.
![open_attach](https://github.com/allure-framework/allure2/assets/24831138/a0f9be80-5d50-4c90-9f01-029a575e8e6b)

### Steps to Reproduce

1. Run test with adding attachment (for example, page source)
2. Generate report
`allure serve target/allure-results`
3. Open test in a report
4. Click 'Open attachment in new tab'

### Expected Behaviour

Attachment opens in new tab

### Screenshots or Additional Context
```
import io.qameta.allure.Allure;
import org.junit.jupiter.api.Test;

import static com.codeborne.selenide.Selenide.open;
import static com.codeborne.selenide.WebDriverRunner.getWebDriver;

class AllureTest {
    public void attachPageSource() {
        String pageSource = getWebDriver().getPageSource();
        if (!pageSource.isEmpty()) {
            Allure.addAttachment("Page source", "text/html", pageSource, "html");
        }
    }

    @Test
    void attachSource() {
        open("http://google.com");
        attachPageSource();
    }
}
```

### What Language are you using?

Java

### What Framework/Allure Integration you are using?

allure-junit5

### What version of Allure Integration you are using?

2.24.0

### What version of Allure Report you are using?

2.24.0

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by baev** [comment]
The behaviour changed due to single file mode support. For single-file reports, the only way to open attachments in the new tab is to use data links. And since all the browsers have limits for URL size, we decided to drop this feature (at least for now). 

We aim to enable single-file mode by default for all users in future releases. Considering this, we prefer to avoid introducing any feature difference between classic & single-file modes. 

However, we are open to hearing your feedback!

**Reply by dkrut** [reply]
It's a pity. I think this is inconvenient when using rest assured steps with attach body, for example...

**Reply by biski** [reply]
For us this is a huge blocker. We have a few attachments like screenshot, logs, rest api logs etc and in case of failed we open them in new tabs. Now we cannot analyze more than two logs at the same time or even analyze a detailed log and compare it to steps from report. The possibility to open an attachment (or more than one) in a new tab is a great advantage of reports.

**Reply by baev** [reply]
We can create a separate report view that renders attachment content in full-screen mode. Will that suite your needs?

**Reply by biski** [reply]
But could we still open attachments in a new tab (for example by click middle mouse button)?

**Reply by baev** [reply]
The way I see it — a special route in the report that will *only* render attachment content and nothing else. You'll be able to open this link in a new browser tab (we can even set target _blank by default).

**Reply by quentinzil** [reply]
I am using this feature a lot, I just recently upgraded to the latest allure version and this is now a problem for me. Would there be a possibility to add an argument to revert back to having attachments open in new tabs if provided when generating a report? Thanks!

**Reply by Nisim761** [reply]
> I am using this feature a lot, I just recently upgraded to the latest allure version and this is now a problem for me. Would there be a possibility to add an argument to revert back to having attachments open in new tabs if provided when generating a report? Thanks!

I also with to have an option to revert back and allow opening attachments in a new browser tab. Please add support for this backward compatibility.

**Comment by biski** [comment]
I've noticed another problem with single-file report. Please take a look at our case. We have a report from weekend tests (100h, 25k test-cases). It contains a lot of test cases. Directory allure-results weighs 1G. When I generate a normal report, it works fast in browser. But when I generate a report in single-file mode and try to open using option "allure open" chrome hangs and finally tab crashes. File index.html weighs 1.5G.

**Reply by baev** [reply]
That's why Allure was implemented as a multi-file report in the first place. Why are you trying to use `single-file` mode for your reports? However, if you want the discussion to continue, please create a separate discussion topic since it's not directly related to this one.

**Comment by quentinzil** [comment]
@baev it is still very inconvenient for my team and myself. Any workaround or anything else we can do other than revert to 2.23? Thanks!

**Reply by baev** [reply]
@quentinzil, could you please describe your use cases so we better understand the issue?

**Reply by quentinzil** [reply]
I usually share the attachment's URL with other members of my team, and/or also sometimes fetches the data using curl. This is not something I can do easily anymore since I have to look through the web page source code to get to the URL.

**Reply by baev** [reply]
We'll be unable to support attachment fetch via curl in single file mode.

**Reply by baev** [reply]
We think of a way to copy & share the link to a step/attachment (similar to how you can share a link to comments/replies in StackOverflow), will it suit your needs?

**Reply by quentinzil** [reply]
I am not generating my reports with the single file mode option, do you mean that this should only be an issue in single file mode?

To clarify my example, this is an url I used to be able to share easily:
[http://my_allure_server.com/my_report/data/attachments/2a9af5d3f2s88108c.txt](url)

Since 2.24, when I click to open attachment in new tab, it downloads it without giving me the chance to copy the url of the attachment itself.
So the only way I have been able to get it is by looking at the page source in the web developer console:
![attachment](https://github.com/user-attachments/assets/d9ef4ba5-e5d8-47ce-924e-4ffb1c0e5b00)

The data in the url above can easily be fetched or downloaded using curl or wget.

Now if there is an easy way for you to implement the copy & share the link to a step/attachment, I believe it will suit my needs. Thanks!

**Reply by baev** [reply]
> do you mean that this should only be an issue in single file mode?

I mean that the behaviour changed when we introduced the single file mode. We can't support the old behaviour in case of single file mode, and we don't think that making it so the same UI element behaves differently when you change the report mode is a good idea. 

From our knowledge, most Allure users would prefer a single file mode to be enabled by default (which we plan to do in Allure 3). Now we are collecting feedback on how users use this feature (and why) to understand use cases better. By doing so, we may come up with a solution that will work for both single file and multi file modes.

**Comment by huxiang** [comment]
If the attachment is a plain text file, opening it in the current tab is totally fine for me.

But if the attachment is a JSON file, opening it in a new browser tab can invoke the browser's JSON parser plugin. It's very efficient for me to do further investigation on it. For example:

<img width="1365" alt="image" src="https://github.com/user-attachments/assets/be86f8f0-76a5-4498-ba81-48ad614974f0" />

Opening the JSON attachment in the current browser tab is really hard to read. Use the same JSON content for example:
<img width="1388" alt="image" src="https://github.com/user-attachments/assets/c571d786-5b62-4202-8846-f9308df93b19" />

<img width="1382" alt="image" src="https://github.com/user-attachments/assets/1bb65dba-e992-415b-ac05-fb2d34305c10" />

-----
Multi-file mode and single file mode are two different user cases, as @biski mentioned, single file mode cannot support large report very well, but multi-file mode can. I think single file mode can be used for simple and small test reports and multi-file mode can be used for complicated and large test reports, it's reasonable to provide better user experience in multi file mode.

**Reply by baev** [reply]
thanks!

**Comment by huxiang** [comment]
Recently I investigated changing the iframe height of embedded html attachment and found a temp solution for it:

After generated the html report with `allure generate` command, open the `<report_dir>/app.js` file, search and remove `,e.setAttribute("download",r)` from the js file. Then downloading attachment behavior should be reverted back: Opening in a new browser tab instead of downloading the file directly.

I'm using the latest release version 2.33.0 of allure, it works perfectly for me. :)

**Comment by Nisim761** [comment]
Will there be an option or a workaround to allow opening an attachment in a new browser tab?
